### PR TITLE
Add GHA lint workflow; Add unit tests; Implement hook options

### PR DESCRIPTION
**Pull request by imnasnainaec**

### CODE REVIEW

File: libs/headless-lfp/src/test/language-font-finder.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { act, renderHook } from '@testing-library/react';
+
+import useLanguageFontFinder from '../language-font-finder';
+
+describe('useLanguageFontFinder', () => {
+  it('should render successfully', async () => {
+    const { result } = renderHook(() => useLanguageFontFinder());
+
+    const {error,finding } = result.current
```
Comment (irahopkinson) [inline libs/headless-lfp/src/test/language-font-finder.spec.tsx:9]: Running `nx format:write` should fix this formatting issue and you can also setup VS Code to do it for you. You might consider running `nx format:check` in `lint.yml` before running the linter.

File: libs/headless-lfp/src/test/utils.spec.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    const fontLFP = convertToFontLFP({
+      ...emptyFontLFF,
+      defaultfamily: mockDefault,
+      families: mockFamilies,
+    });
+    expect(fontLFP.name).toEqual(mockName);
+  });
+});
+
+describe('fetchJson', () => {
```
Comment (irahopkinson) [inline libs/headless-lfp/src/test/utils.spec.ts:46]: nit: Strictly testing the external API isn't a unit test but more like an integration test but I expect it's useful to have this for now.

**Review by irahopkinson — APPROVED** [review]
👍 LGTM

File: libs/headless-lfp/src/test/language-font-finder.spec.tsx (hunk 1 above)
Comment (imnasnainaec) [reply libs/headless-lfp/src/test/language-font-finder.spec.tsx:9]: Good call. Coming up!

**Pull request merged**